lib/qam.pm

```diff
@@ -136,10 +136,16 @@ sub get_patches {
     $repo =~ tr/,/ /;
 
     # Search for patches by incident, exclude not needed
-    my $patches = script_output("zypper patches -r $repo | awk -F '|' '/[Nn]eeded/ && !/[Nn]ot [Nn]eeded/ && /$incident_id/ { printf \$2 }'");
-    # Remove carriage returns and make patch list on one line
-    $patches =~ s/\r//g;
-    return $patches;
+    my $patches = script_output("zypper patches -r $repo");
+    my @patch_list;
+
+    for my $line (split /\n/, $patches) {
+        my @tokens = split /\s*\|\s*/, $line;
+        next if scalar @tokens < 6;
```

Comment:
What is this magic value?

Reply:
I guess it's number of columns in patches table. That detect if it's a row of table with patches or not.

Reply:
> What is this magic value?

The next line reads `$tokens[5]` so this is a check that it exists. I prefer to check that my indices are within range even when it's redundant.

Reply:
I am not against index checking, I just like to know, what the non-zero/non-one constants are. Wouldn't be better to replace the number with a proper constant to clarify the meaning (the main idea of this line went away with removing of the `awk` from the pipe)?

Reply:
I'm usually using some sort of constants for things like this as well, but didn't want to bother Martin about this.

Reply:
Code updated to use constants. I can't create a new verification run on OSD because all the kgraft incidents have been purged but it works on my local OpenQA instance (with a short custom test module).